[merged] HifContext: set error if os-release not found

jlebon opened the pull request.

cgwalters commented on libhif/hif-context.c:
```diff
@@ -997,8 +997,13 @@ hif_context_set_os_release(HifContext *context, GError **error)
+        if (!g_file_get_contents(os_release, &contents, NULL, NULL)) {
+            g_set_error(error,
+                        HIF_ERROR,
+                        HIF_ERROR_FILE_NOT_FOUND,
```
Not necessarily...we could get EIO or EPERM etc.  So wouldn't it be simpler to just pass `error` instead of `NULL` as the last arg to `g_file_get_contents()` ?

jlebon replied:
True, but then it will probably only list /usr/lib/os-release in the error msg instead of a more generic one. Though I can explicitly check for G_FILE_ERROR_NOENT for that. Will rework.

jlebon commented:
Added a fixup! I just made it handle and report both errors individually.

ignatenkobrain commented:
hey @jlebon, why don't we use here `g_autoptr()` for `GError`?

cgwalters commented on libhif/hif-context.c:
```diff
@@ -992,20 +992,30 @@ hif_context_set_os_release(HifContext *context, GError **error)
     }
 
+    if (etc_error != NULL)
+        g_error_free (etc_error);
```
Could use `g_autoptr` for the errors.

cgwalters commented:
I think the cleanest would be importing `ot_gfile_load_contents_utf8_allow_noent()` - if we hit an error other than `ENOENT` I think we typically want to throw rather than looking for another file.  That also makes it easier to just pass through the error rather than wrapping it.

(I'm not sure there's value in re-wrapping with `HIF_ERROR` - I don't typically do that in my projects since I think cases where people want to distinguish on error codes are relatively rare and generally better handled in another way)

cgwalters commented:
`git grep allow_noent` in ostree for example for other instances of this pattern.

jlebon commented:
:arrow_up: Updated. (I just rewrote the branch since the interdiff was getting silly).

I still wanted a customized error msg when both were not found to indicate we tried both locations, so that made the patch a bit beefier.

jlebon commented:
@cgwalters WDYT of the latest commit? :nut_and_bolt:

cgwalters commented:
Oops, I missed your followup comment that it'd been updated.  LGTM, and sorry about the bikeshedding on this but error handling is important I think.

@rh-atomic-bot r+ 1125957
